log4j-core/src/main/java/org/apache/logging/log4j/core/config/ConfigurationSource.java

```diff
@@ -339,17 +339,9 @@ public static ConfigurationSource fromUri(final URI configLocation) {
             return null;
         }
         try {
-            URL url = configLocation.toURL();
-            URLConnection urlConnection = UrlConnectionFactory.createConnection(url);
-            InputStream is = urlConnection.getInputStream();
-            long lastModified = urlConnection.getLastModified();
-            return new ConfigurationSource(is, configLocation.toURL(), lastModified);
-        } catch (final FileNotFoundException ex) {
-            ConfigurationFactory.LOGGER.warn("Could not locate file {}", configLocation.toString());
+            return getConfigurationSource(configLocation.toURL());
```

Comment:
The replacement protects against dangling file descriptors that appear when `getLastModified()` is called.